core/src/main/scala/cats/Unfoldable.scala

```diff
@@ -0,0 +1,59 @@
+package cats
+
+import simulacrum.typeclass
+
+@typeclass trait Unfoldable[F[_]] { self =>
+
+  /**
+    * Build an F[A] from a `seed` and a generating function `f``
```

Comment:
it would be nice to have some stronger laws, even that might relate `Unfoldable` to `Foldable` or `Functor`.

Such as, if `F[_]: Unfoldable` and `F[_]: Functor` then,
```scala
unfoldLeft(seed)(fn).map(g) == unfoldLeft(seed)(fn.andThen { case (b, a) => (b, g(a)) })`
```